### DISCUSSION TITLE
Utils vs String split

### DISCUSSION THREAD

**Discussion by devgianlu** [librespot-org/librespot-java#276] [Q&A] [answered]
So... Just trying to understand how this thing work. @Iscle asked the question and I marking it as the correct answer even if the answer is my comment to his question.

**Comment by iscle** [comment, marked as answer]
Hi!

This section is great.

I was reading through the code, and I can't see what the difference between Utils.split and String.split functions is.

The Utils.split function is here: https://github.com/librespot-org/librespot-java/blob/master/lib/src/main/java/xyz/gianlu/librespot/common/Utils.java#L133

I'm wondering if it has a special feature not present in the String.split function, or it's just an old piece of code that has sticked...

Thanks!

**Reply by devgianlu** [reply]
The only difference with String.split is that it splits on a character instead of a regex string. For what I remember, I introduced it for performance reasons, but I don't know if it is faster than the regex implementation in a significant manner.

**Reply by devgianlu** [reply]
Can't understand how this section works, it allows me to pick an answer, but why should I. Does it let you?
